numpy/distutils/fcompiler/gnu.py

```diff
@@ -159,10 +146,12 @@ def get_library_dirs(self):
                 # if windows and not cygwin, libg2c lies in a different folder
                 if sys.platform == 'win32' and not d.startswith('/usr/lib'):
                     d = os.path.normpath(d)
-                    if not os.path.exists(os.path.join(d, "lib%s.a" % self.g2c)):
-                        d2 = os.path.abspath(os.path.join(d,
-                                                          '../../../../lib'))
-                        if os.path.exists(os.path.join(d2, "lib%s.a" % self.g2c)):
+                    path = os.path.join(d, "lib%s.a" % self.g2c)
+                    if not os.path.exists(path):
+                        d2 = os.path.join(d, '../../../../lib')
```

Comment:
Even though it works as is, this is probably best done as `join(d, '..', '..', '..'. '..'. 'lib')`, just in case Windows decides to start barfing on `/` as a directory separator....